CONTRIBUTING.md

```diff
@@ -302,92 +338,183 @@ into any of those categories, use your best judgement.
-to `types-foo-x.y.n+1`.
+This test requires Python 3.5 or higher; Python 3.6.1 or higher is recommended.
+Run using:
+```
+(.venv3)$ python3 tests/mypy_test_suite.py
+```
 
-### What to do when a project's documentation and implementation disagree
+This test runs mypy's own test suite using the typeshed code in your repo. This
+will sometimes catch issues with incorrectly typed stubs, but is much slower
+than the other tests.
 
-Type stubs are meant to be external type annotations for a given
-library.  While they are useful documentation in its own merit, they
-augment the project's concrete implementation, not the project's
-documentation.  Whenever you find them disagreeing, model the type
-information after the actual implementation and file an issue on the
-project's tracker to fix their documentation.
+### check\_consistent.py
+
+Run using:
+```
+python3 tests/check_consistent.py
+```
 
-## Issue-tracker conventions
+### stubtest\_test.py
 
-We aim to reply to all new issues promptly.  We'll assign one or more
-labels to indicate we've triaged an issue, but most typeshed issues
-are relatively simple (stubs for a given module or package are
-missing, incomplete or incorrect) and we won't add noise to the
-tracker by labeling all of them.  Please see the
-[list of all labels](https://github.com/python/typeshed/issues/labels)
-for a detailed description of the labels we use.
+This test requires Python 3.6 or higher.
+Run using
+```
+(.venv3)$ python3 tests/stubtest_test.py
+```
 
-Sometimes a PR can't make progress until some external issue is
-addressed.  We indicate this by editing the subject to add a ``[WIP]``
-prefix.  (This should be removed before committing the issue once
-unblocked!)
+This test compares the stdlib stubs against the objects at runtime. Because of
+this, the output depends on which version of Python and on what kind of system
+it is run.
+Thus the easiest way to run this test is via Github Actions on your fork;
+if you run it locally, it'll likely complain about system-specific
+differences (in e.g, `socket`) that the type system cannot capture.
+If you need a specific version of Python to repro a CI failure,
+[pyenv](https://github.com/pyenv/pyenv) can help.
+
+Due to its dynamic nature, you may run into false positives. In this case, you
+can add to the whitelists for each affected Python version in
+`tests/stubtest_whitelists`. Please file issues for stubtest false positives
+at [mypy](https://github.com/python/mypy/issues).
+
+To run stubtest against third party stubs, it's easiest to use stubtest
+directly, with
+```
+(.venv3)$ python3 -m mypy.stubtest \
+  --custom-typeshed-dir <path-to-typeshed> \
+  <third-party-module>
+```
+stubtest can also help you find things missing from the stubs.
 
-### Maintainer guidelines
+
+### flake8
+
+flake8 requires Python 3.6 or higher. Run using:
+```
+(.venv3)$ flake8
+```
+
+Note typeshed uses the `flake8-pyi` and `flake8-bugbear` plugins.
+
+## Submitting Changes
```

Comment:
I'd move this section higher

Reply:
The order follows the order of the steps in the first section (find place, prepare changes, optionally test, submit changes). I'd like to keep the order of the chapters the same.